### PR TITLE
Added support for Alda

### PR DESCRIPTION
**Pull request by stellarskylark**
Created a lexer, spec, and example files for the Alda music composition language.

Language details here: [Alda Docs](https://github.com/alda-lang/alda/blob/master/doc/index.md)

### CODE REVIEW

File: lib/rouge/lexers/alda.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      tag 'alda'
+      filenames "*.alda"
+      mimetypes 'text/alda'
+
+      def self.detect?(text)
+        return true if text.shebang? 'alda'
+      end
+
+      state :root do
+        rule /#[ \S]+/, Comment
```
Comment (pyrmont) [inline lib/rouge/lexers/alda.rb]: I only had a very quick look at the docs so I may have missed it but does the hash mark really need to be followed by a space and not any other form of whitespace?

File: lib/rouge/lexers/alda.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        rule /(\w+)( "[[:alpha:]]{2}[\w\-\+'()]*"+)?(:)/ do |m|
+          token Keyword::Type, m[1]
+          token Name::Label, m[2]
+          token Punctuation, m[3]
+        end
+
+        rule /(\()([\w\-]+!?)( )([^\)]+)(\))/ do |m|
+          token Punctuation, m[1]
+          token Keyword::Reserved, m[2]
+          token Text, m[3]
```
Comment (pyrmont) [inline lib/rouge/lexers/alda.rb]: Probably more correct to use the `Text::Whitespace` token here.

**Comment by pyrmont** [comment]
@astroengisci Thanks for taking the time to submit a lexer! Sorry it's taken so long to reply :( Have submitted a review with some comments on the lexer (which looks pretty good!).

**Comment by UlyssesZh** [comment]
Hi @stellarskylark, why did you close this pull request?

**Pull request closed without merging**